waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -69,8 +84,107 @@ analyzeWaspProject waspDir options = do
   where
     fileNotFoundMessage = "Couldn't find the *.wasp file in the " ++ toFilePath waspDir ++ " directory"
 
-analyzeWaspFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
-analyzeWaspFile prismaSchemaAst waspFilePath = do
+data WaspFile
+  = WaspLang !(Path' Abs (File WaspLangFile))
+  | WaspTs !(Path' Abs (File WaspTsFile))
+
+data WaspLangFile
+
+data WaspTsFile
+
+data CompiledWaspJsFile
+
+data DeclsJsonFile
+
+analyzeWaspFile :: Path' Abs (Dir WaspProjectDir) -> Psl.Schema.Schema -> WaspFile -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspFile waspDir prismaSchemaAst = \case
+  WaspLang waspFilePath -> analyzeWaspLangFile prismaSchemaAst waspFilePath
+  WaspTs waspFilePath -> analyzeWaspTsFile waspDir prismaSchemaAst waspFilePath
+
+analyzeWaspTsFile :: Path' Abs (Dir WaspProjectDir) -> Psl.Schema.Schema -> Path' Abs (File WaspTsFile) -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspTsFile waspProjectDir prismaSchemaAst waspFilePath = runExceptT $ do
+  -- TODO: I'm not yet sure where tsconfig.node.json location should come from
+  -- because we also need that knowledge when generating a TS SDK project.
+  compiledWaspJsFile <- ExceptT $ compileWaspTsFile waspProjectDir [relfile|tsconfig.node.json|] waspFilePath
+  declsJsonFile <- ExceptT $ executeMainWaspJsFile waspProjectDir prismaSchemaAst compiledWaspJsFile
+  ExceptT $ readDeclsFromJsonFile declsJsonFile
+
+compileWaspTsFile ::
+  Path' Abs (Dir WaspProjectDir) ->
+  Path' (Rel WaspProjectDir) File' ->
+  Path' Abs (File WaspTsFile) ->
+  IO (Either [CompileError] (Path' Abs (File CompiledWaspJsFile)))
+compileWaspTsFile waspProjectDir tsconfigNodeFileInWaspProjectDir waspFilePath = do
+  chan <- newChan
+  (_, tscExitCode) <-
+    concurrently
+      (readJobMessagesAndPrintThemPrefixed chan)
+      ( runNodeCommandAsJob
```

Comment:
Maybe stupid question, because I forgot a bit about this how exactly `runNodeCommandAsJob` is used in the Generator event hough I wrote it, but why do we even need it here? Because it checks node version? Or because of its capabilities to read job messages and print them prefixed? But do we need the whole concept of Job here at all? And prefixing of messages? We need it in Generator because we are multiplexing the outputs from multiple Jobs if I remember correctly, but I don't think we are multiplexing this with anything?

Reply:
We need to prefix the messages, yes (will become clear why when you run this in the next PR).

Do we need a Job? Probably.
There's no multiplexing, but we want the `[Wasp]` output, and this is how we did it when installing Wasp NPM dependencies (also no multiplexing). I figured this command was the standard for running any node or npm-related command under a specific message prefix.

Does that make sense?

Reply:
So 'trust me bro' :D. Kidding -> you need prefixes for some reason + you do like that it ensures node version. Ok, good enough for me.